components/storage/blockdevice/COMPONENT_I2CEE/I2CEEBlockDevice.cpp

```diff
@@ -68,7 +68,9 @@ int I2CEEBlockDevice::read(void *buffer, bd_addr_t addr, bd_size_t size)
 
     _i2c->stop();
 
-    if (_i2c->read(_i2c_addr, static_cast<char *>(buffer), size) < 0) {
+    _sync();
```

Comment:
Should we check the error code from `_sync()` as well?

Reply:
Good catch, will do this this evening. Thx.

Reply:
Update is out.